scripts/core/ui/components/ListPage/generic-list-page-item-view-edit.tsx

```diff
@@ -87,13 +88,17 @@ export class GenericListPageItemViewEdit<T> extends React.Component<IProps<T>, I
 
     handleFieldChange(field: string, nextValue: valueof<IProps<T>['item']>) {
         // using updater function to avoid race conditions
-        this.setState((prevState) => ({
-            ...prevState,
-            nextItem: {
-                ...prevState.nextItem,
-                [field]: nextValue,
-            },
-        }));
+
+        this.setState((prevState) => {
+            const nextItem = {...prevState.nextItem};
+
+            set(nextItem, field, nextValue);
```

Comment:
`lodash.set` mutates and react doesn't re-render if mutated because it uses reference equality to decide whether to re-render. Deep cloning would work around it by always re-rendering which ensures correctness, but messes up performance. Using Immer library would be a proper solution here.